karapace/schema_backup.py

```diff
@@ -311,19 +313,23 @@ def restore_backup(self) -> None:
                     self._restore_backup_version_2(producer, fp)
                 else:
                     self._restore_backup_version_1_single_array(producer, fp)
+            producer.flush(timeout=self.timeout_kafka_producer)
+            if self.producer_exception:
```

Comment:
We have to be explicit about what we do, otherwise some weird Python coercion is going to bite us in the future. 😉

```suggestion
            if self.producer_exception is not None:
```